### ISSUE TITLE
Menus from DBeaver appear on every workspace

### ISSUE THREAD

**Issue by mihalyr** [dbeaver/dbeaver#36155]
### Description

Open a menu on DBeaver

![image](https://github.com/user-attachments/assets/27f3908f-7fad-4c9f-9729-98d3baff94f6)

The menu shows up on unrelated workspaces everywhere and can only be closed when returning to DBeaver workspace and closing it there. 

![image](https://github.com/user-attachments/assets/d9f7ac40-528e-4979-be25-d54502dbf985)

Sometimes this happens also with tooltips

### DBeaver Version

Community 23.2.4

### Operating System

Fedora 41 Sericea (Sway/Wayland)

### Database and driver

N/A

### Steps to reproduce

1. Open DBeaver
2. Open a menu from the menubar
3. Switch to a different workspace
4. Menu bar shows up on unrelated workspaces

**Comment by mihalyr** [comment]
Or sometimes this happens also with tooltips, see below a randomly appearing tooltip from DBeaver showing up on a different workspace from where the app is running. I could not find out how to replicate this for tooltips, they just happen once in a while...

![image](https://github.com/user-attachments/assets/df3c1aa2-e9ce-473d-8907-1b2dc1198f89)

**Comment by mihalyr** [comment]
I switched to the Flatpak version of 24.2.3 and I cannot reproduce this. But I noticed that the Flatpak version now runs natively under Wayland without XWayland, so this might be actually working now. I haven't tried the latest non-flatpak without XWayland, so not sure if that works also - the only reason I avoided the Flatpak were some past issues:
* https://github.com/flathub/io.dbeaver.DBeaverCommunity/issues/158
* https://github.com/flathub/io.dbeaver.DBeaverCommunity/issues/156
But I'm going to give it another try now, maybe I'll mask it from updates so it doesn't break. Closing this for now, it seems the latest flatpak does not has this problem anymore.
